rma/models/rma.py

```diff
@@ -819,25 +871,28 @@ def action_preview(self):
             "url": self.get_portal_url(),
         }
 
-    def action_view_receipt(self):
-        """Invoked when 'Receipt' smart button in rma form view is clicked."""
+    def _action_view_pickings(self, pickings):
         self.ensure_one()
         # Force active_id to avoid issues when coming from smart buttons
         # in other models
-        action = (
-            self.env.ref("stock.action_picking_tree_all")
-            .sudo()
-            .with_context(active_id=self.id)
-            .read()[0]
-        )
-        action.update(
-            res_id=self.reception_move_id.picking_id.id,
-            view_mode="form",
-            view_id=False,
-            views=False,
+        action = self.env["ir.actions.act_window"]._for_xml_id(
+            "stock.action_picking_tree_all"
         )
+        if len(pickings) > 1:
+            action["domain"] = [("id", "in", pickings.ids)]
+        elif pickings:
+            action.update(
+                res_id=pickings.id,
+                view_mode="form",
+                view_id=False,
+                views=False,
+            )
         return action
 
+    def action_view_receipt(self):
+        """Invoked when 'Receipt' smart button in rma form view is clicked."""
+        return self._action_view_pickings(self.reception_move_id.picking_id)
```

Comment:
Shouldn't this cover the possibility of having the RMA receipt in several steps?

Reply:
We are not adding this for the moment. It's just that this method to be prepared when that happens (if ever). Add in ROADMAP the sentence "Add RMA reception and/or RMA delivery on several steps - 2 or 3 - like normal receptions/deliveries. It should be a separate option inside the warehouse definition.".

That's because you can have different flows for regular moves and RMA ones.